include/libopencm3/stm32/l1/ri.h

```diff
@@ -0,0 +1,658 @@
+#define RI_CMR1                 MMIO32(RI_BASE + 0x24)
+#define RI_CICR1                MMIO32(RI_BASE + 0x28)
+#define RI_ASMR2                MMIO32(RI_BASE + 0x2c)
+#define RI_CMR2                 MMIO32(RI_BASE + 0x30)
+#define RI_CICR2                MMIO32(RI_BASE + 0x34)
+#define RI_ASMR3                MMIO32(RI_BASE + 0x38)
+#define RI_CMR3                 MMIO32(RI_BASE + 0x3c)
+#define RI_CICR3                MMIO32(RI_BASE + 0x40)
+#define RI_ASMR4                MMIO32(RI_BASE + 0x44)
+#define RI_CMR4                 MMIO32(RI_BASE + 0x48)
+#define RI_CICR4                MMIO32(RI_BASE + 0x4c)
+#define RI_ASMR5                MMIO32(RI_BASE + 0x50)
+#define RI_CMR5                 MMIO32(RI_BASE + 0x54)
+#define RI_CICR5                MMIO32(RI_BASE + 0x58)
+
+/**
+ * RI input capture register
+ *
+ * The RI_ICR register is used to select the routing of 4 full ports to the
+ * input captures of TIM2, TIM3 and TIM4.
+ */
+#define RI_ICR_IC1IOS_SHIFT     0
+#define RI_ICR_IC1IOS_MASK      0xf
+#define RI_ICR_IC2IOS_SHIFT     4
+#define RI_ICR_IC2IOS_MASK      0xf
+#define RI_ICR_IC3IOS_SHIFT     8
+#define RI_ICR_IC3IOS_MASK      0xf
+#define RI_ICR_IC4IOS_SHIFT     12
+#define RI_ICR_IC4IOS_MASK      0xf
+#define RI_ICR_TIM_SHIFT        16
+#define RI_ICR_TIM_MASK         0x3
+#define RI_ICR_IC1              (1 << 18)
+#define RI_ICR_IC2              (1 << 19)
+#define RI_ICR_IC3              (1 << 20)
+#define RI_ICR_IC4              (1 << 21)
+
+/**
+ * RI analog switches control register 1
+ *
+ * The RI_ASCR1 register is used to configure the analog switches of the I/Os
+ * linked to the ADC. These I/Os are pointed to by the ADC channel number.
+ */
+#define RI_ASCR1_GR1_1          (1 << 0)
+#define RI_ASCR1_GR1_2          (1 << 1)
+#define RI_ASCR1_GR1_3          (1 << 2)
+#define RI_ASCR1_GR1_4          (1 << 3)
+#define RI_ASCR1_CH4            (1 << 4)
+#define RI_ASCR1_CH5            (1 << 5)
+#define RI_ASCR1_GR2_1          (1 << 6)
+#define RI_ASCR1_GR2_2          (1 << 7)
+#define RI_ASCR1_GR3_1          (1 << 8)
+#define RI_ASCR1_GR3_2          (1 << 9)
+#define RI_ASCR1_GR8_1          (1 << 10)
+#define RI_ASCR1_GR8_2          (1 << 11)
+#define RI_ASCR1_GR8_3          (1 << 12)
+#define RI_ASCR1_GR8_4          (1 << 13)
+#define RI_ASCR1_GR9_1          (1 << 14)
+#define RI_ASCR1_GR9_2          (1 << 15)
+#define RI_ASCR1_GR11_5         (1 << 16)
+#define RI_ASCR1_GR7_1          (1 << 18)
+#define RI_ASCR1_GR7_2          (1 << 19)
+#define RI_ASCR1_GR7_3          (1 << 20)
+#define RI_ASCR1_GR7_4          (1 << 21)
+#define RI_ASCR1_CH22           (1 << 22)
+#define RI_ASCR1_CH23           (1 << 23)
+#define RI_ASCR1_CH24           (1 << 24)
+#define RI_ASCR1_CH25           (1 << 25)
+#define RI_ASCR1_VCOMP          (1 << 26)
+#define RI_ASCR1_GR11_1         (1 << 27)
+#define RI_ASCR1_GR11_2         (1 << 28)
+#define RI_ASCR1_GR11_3         (1 << 29)
+#define RI_ASCR1_GR11_4         (1 << 30)
```

Comment:
For consistency with the ref manual, I'd rather see these as CH5, CH4, and CH3_GR1_4, CH2_GR1_3.  I understand how gross it is, but it at least makes all the definitions here consistent with the ref man names, gross as they are.